demo/lightning/checkpoint/simulated/multiprocessing_train.py

```diff
@@ -268,12 +198,14 @@ def time_checkpoint_operation(benchmark_strategy: BenchmarkStrategy,
     saving/loading under distributed settings.
     """
     times = []
-    template_state_dict = model.state_dict()
-    for key, tensor in template_state_dict.items():
-        template_state_dict[key] = torch.empty_like(tensor)
-    if hasattr(model, 'dummy_tensors'):
-        for i, tensor in enumerate(model.dummy_tensors):
-            template_state_dict[f'dummy_tensor_{i}'] = torch.empty_like(tensor)
+    template_state_dict = dict()
+    # According to `create_default_local_load_plan` https://github.com/pytorch/pytorch/blob/main/torch/distributed/checkpoint/default_planner.py#L343
+    # each key will be read only once from the state_dict, hence assigning different names to different tensor will force the load function to only read
+    # tensor shard corresponding to given node.
+    for i in range(tensor_count):
+        if i % world_size == rank:
+            template_state_dict[f'dummy_tensor_{i}'] = torch.empty(
+                tensor_size, 1000)
     for i in range(sample_count):
         checkpoint_path = os.path.join(filepath, f'checkpoints/ckpt_{i}.ckpt')
         dist.barrier()
```

Comment:
Should this dist.barrier() call be between the two for loops? Do we need it inside the loop given we already have a barrier call below after the save/load combo?

Reply:
Not really, I want all the nodes to start and end the save/load checkpoint at the same time, so that we can see if number of concurrent requests to GCS causes any regressions in the throughput. 

I dont see any problem if dist.barrier() is not present between the for loop, since the names of tensors are different.